src/client/api/container.c

```diff
@@ -100,7 +53,7 @@ cont_inherit_redunc_fac(daos_handle_t poh, daos_prop_t *cont_prop,
 		*merged_prop = daos_prop_merge(cont_prop, redunc_prop);
 		daos_prop_free(redunc_prop);
 		if (*merged_prop == NULL) {
-			D_ERROR("failed to merge cont_prop and redunc_prop\n");
+			D_ERROR("failed to merge cont_prop and redunc_prop "DF_RC"\n", DP_RC(-DER_NOMEM));
 			rc = -DER_NOMEM;
```

Comment:
I think this would be slightly cleaner.
```suggestion
			rc = -DER_NOMEM;
			D_ERROR("failed to merge cont_prop and redunc_prop "DF_RC"\n", DP_RC(rc));
```
It looks like there are several ways daos_prop_merge could fail so -DER_NOMEM is an assumption here but a reasonable one.  Unfortunately it looks like it's part of the public API so we can't really change it.

Reply:
I agree. Thank you! I will change it in next commit. Right. I checked daos_prop_merge() and several functions called inside this function. Normally, -DER_NOMEM may be the most likely failure. Under certain circumstances like buffer overflow, the assumption could be wrong and the code here could report false error code.